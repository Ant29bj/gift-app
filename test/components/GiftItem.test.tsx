import { render, screen } from "@testing-library/react"
import GiftItem from "../../src/components/GifItem"
import 'jest-extended';

describe('Test on <GiftItem/>', () => {
  const title = 'mock title'
  const url = 'mock url'
  test('should match with the snapshot', () => {
    const { container } = render(<GiftItem title={title} url={url} />)
    expect(container).toMatchSnapshot();
  });

  test('should show the image with the title and url', () => {
    render(<GiftItem title={title} url={url} />)
    const img = screen.getByRole('img')
    expect(img.getAttribute('src')).toBe(url)
    expect(img.getAttribute('alt')).toBe(title)
  })

  test('should print the title', () => {
    render(<GiftItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy();
  })
})