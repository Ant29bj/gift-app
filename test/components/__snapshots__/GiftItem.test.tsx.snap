// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test on <GiftItem/> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      mock title
    </p>
    <img
      alt="mock title"
      src="mock url"
    />
  </div>
</div>
`;
